src/SoapCore/MessageEncoder/SoapMessageEncoder.cs

```diff
@@ -141,9 +141,23 @@
 				throw new ArgumentNullException(nameof(stream));
 			}
 
-			XmlReader reader = _supportXmlDictionaryReader ?
-			 	XmlDictionaryReader.CreateTextReader(stream, _writeEncoding, ReaderQuotas, dictionaryReader => { }) :
-				XmlReader.Create(stream, new XmlReaderSettings() { IgnoreWhitespace = true, DtdProcessing = DtdProcessing.Prohibit });
+			XmlReader reader;
+			if (_supportXmlDictionaryReader)
+			{
+				reader = XmlDictionaryReader.CreateTextReader(stream, _writeEncoding, ReaderQuotas, dictionaryReader => { });
+			}
+			else
+			{
+				var xmlReaderSettings = new XmlReaderSettings()
+				{
+					IgnoreWhitespace = true,
+					DtdProcessing = DtdProcessing.Prohibit,
+					CloseInput = true
+				};
+
+				var streamReaderWithEncoding = new StreamReader(stream, _writeEncoding);
+				reader = XmlReader.Create(streamReaderWithEncoding, xmlReaderSettings);
```

Comment:
Hmm, looks like a false positive, because `DtdProcessing = DtdProcessing.Prohibit` is set a few lines above. And in fact it was already here before my changes.